src/inference/src/dev/core_impl_ie.cpp

```diff
@@ -209,7 +192,12 @@ InferenceEngine::RemoteContext::Ptr ov::CoreImpl::CreateContext(const std::strin
  */
 void ov::CoreImpl::AddExtension(const InferenceEngine::IExtensionPtr& extension) {
     std::lock_guard<std::mutex> lock(get_mutex());
-    AddExtensionUnsafe(extension);
+    try {
+        AddExtensionUnsafe(extension);
+    } catch (const ov::Exception& ex) {
+        // wrap to legacy exception
+        throw InferenceEngine::Exception(ex.what());
```

Comment:
why do we wrap this place not don't wrap others? It's OLD API and it should throw old exceptions.
I suppose all methods in current file should throw old exceptions